### ISSUE TITLE
USB Pi Zero keyboard

### ISSUE THREAD

**Issue by billybicep** [GideonZ/1541ultimate#184]
I have a 3 key USB keyboard which is hosted on a Raspberry Pi Zero, this enables you to map any key to the 3 keys. I have these mapped as F10 and F11 to initiate Freeze or the Menu. The keys work as mapped when on any windows machine but dont work when i plug it into the USB on my Ultimate 64. Is this due to the U64 needing a HID driver as mentioned in issue #171 ?

Thanks.

**Comment by GideonZ** [comment]
Yes, that is precisely correct. You can also change your USB descriptors to
report a BOOTMODE keyboard that uses a standard 8-byte report. Then it will
work.

**Comment by billybicep** [comment]
OK thanks Gideon, I'll ask the Pimoroni team (they make the device) if they can help, i have no idea when it gets this technical!

**Comment by uffejakobsen** [comment]
@billybicep My best guess is that you should change "something" here https://github.com/pimoroni/keybow-firmware/blob/master/keybow/gadget-hid.c
But my usb descriptor knowledge is a little dusty and needs a brush-up
Please keep us (me) posted on the progress with the Piramoni folks :-)

**Comment by billybicep** [comment]
looks like the setup is all compiled into a .img file, i'll see what Pimoroni come back with.

**Comment by billybicep** [comment]
No response from Pimoroni, no response from the pi Forums other than someone who seems to think i'm asking for the holy grail. I'll leave this until hopefully the U64 has a HID driver. Thanks.
